aten/src/ATen/native/cudnn/BatchNorm_v8.cpp

```diff
@@ -0,0 +1,911 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Config.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/cuda/CUDAConfig.h>
+
+#ifdef __HIP_PLATFORM_AMD__
+#include <ATen/native/cudnn/hip/BatchNorm.h>
+#else
+#include <ATen/native/cudnn/BatchNorm.h>
+#endif
+
+#include <ATen/native/ConvUtils.h> // for cudnnv8_enabled_check_debug()
+
+#if AT_CUDNN_ENABLED()
+
+#include <ATen/cudnn/cudnn-wrapper.h>
+#include <c10/macros/Macros.h>
+
+C10_DIAGNOSTIC_PUSH_AND_IGNORED_IF_DEFINED("-Wsuggest-override")
+#include <cudnn_frontend.h>
+C10_DIAGNOSTIC_POP()
+
+#include <ATen/TensorUtils.h>
+#include <ATen/cuda/Exceptions.h>
+#include <ATen/cudnn/Handle.h>
+#include <ATen/native/utils/ParamsHash.h>
+#include <c10/cuda/CUDACachingAllocator.h>
+#include <c10/cuda/CUDAException.h>
+#include <c10/util/env.h>
+
+#include <list>
+#include <unordered_map>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/cudnn_batch_norm_backward_native.h>
+#include <ATen/ops/cudnn_batch_norm_native.h>
+#include <ATen/ops/empty.h>
+#include <ATen/ops/empty_like.h>
+#endif
+
+namespace at {
+namespace native {
+
+namespace {
+namespace fe = cudnn_frontend;
+
+auto get_fe_dtype(const Tensor& t) {
+  auto dtype = t.scalar_type();
+  auto fe_dtype = fe::DataType_t::FLOAT;
```

Comment:
Why specify a default fe_dtype at all? Is it needed for auto I guess? Wouldn't this be a text book example of switch statement handling btw?